### PR TITLE
Minor tweaks to build process

### PR DESCRIPTION
**Pull request by paulfantom**
- Make sure go modules are always enabled for better developer experience
- Use go1.12 in container image

/cc @brancz @s-urbaniak

### CODE REVIEW

**Comment by paulfantom** [comment]
Seems like CI failure is caused by a network problem. Retriggereing by a force-push as I don't have repository permissions (cc @brancz)

**Comment by paulfantom** [comment]
Apparently it was a problem with an intermediary build image lacking `git`. Added in last commit.

**Comment by brancz** [comment]
lgtm :+1: 

invite sent

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-FROM golang:1.11-alpine AS build
-RUN apk add --update make
+FROM golang:1.12-alpine AS build
+RUN apk add --update make && apk add --no-cache git
```
Comment (metalmatze) [inline Dockerfile]: FYI, we could probably do everything in one command. 
```diff
- RUN apk add --update make && apk add --no-cache git
+ RUN apk add --update --no-cache make git
```
Nothing to change now though.

**Pull request merged**